src/main/java/net/earthcomputer/clientcommands/command/ConnectFourCommand.java

```diff
@@ -0,0 +1,364 @@
+                for (int y = 3; y < HEIGHT; y++) {
+                    if ((this.board[x][y] & this.board[x + 1][y - 1] & this.board[x + 2][y - 2] & this.board[x + 3][y - 3]) > 0) {
+                        return this.board[x][y];
+                    }
+                }
+            }
+
+            for (int x = 0; x < WIDTH; x++) {
+                for (int y = 0; y < HEIGHT; y++) {
+                    if (this.board[x][y] == 0) {
+                        // still a space to play
+                        return 0;
+                    }
+                }
+            }
+
+            // no spaces left, game ends in a draw
+            return Piece.RED | Piece.YELLOW;
+        }
+
+        public static boolean isValidRow(int x) {
+            return 0 <= x && x < WIDTH;
+        }
+
+        public int getPlacementY(int x) {
+            int y = 0;
+            for (byte piece : this.board[x]) {
+                if (piece == 0) {
+                    break;
+                }
+                y++;
+            }
+
+            return y;
+        }
+    }
+
+    public static class Piece {
+        public static final byte RED = 1 << 0;
+        public static final byte YELLOW = 1 << 1;
+
+        public static byte opposite(byte piece) {
+            return switch (piece) {
+                case RED -> YELLOW;
+                case YELLOW -> RED;
+                default -> throw new IllegalStateException("Unexpected value: " + piece);
+            };
+        }
+
+        public static Component name(byte piece) {
+            return switch (piece) {
+                case RED -> Component.translatable("connectFourGame.pieceRed");
+                case YELLOW -> Component.translatable("connectFourGame.pieceYellow");
+                default -> throw new IllegalStateException("Unexpected value: " + piece);
+            };
+        }
+
+        public static void render(GuiGraphics graphics, int x, int y, byte piece, boolean transparent) {
+            if (piece == 0) {
+                return;
+            }
+            int xOffset = switch (piece) {
+                case RED -> 0;
+                case YELLOW -> 16;
+                default -> throw new IllegalStateException("Unexpected value: " + piece);
+            };
+            graphics.blit(
+                RenderType::guiTextured,
+                ConnectFourGameScreen.PIECES_TEXTURE,
+                x,
+                y,
+                xOffset,
+                0,
+                ConnectFourGameScreen.PIECE_WIDTH,
+                ConnectFourGameScreen.PIECE_HEIGHT,
+                ConnectFourGameScreen.TEXTURE_PIECE_WIDTH,
+                ConnectFourGameScreen.TEXTURE_PIECE_HEIGHT,
+                ConnectFourGameScreen.TEXTURE_PIECES_WIDTH,
+                ConnectFourGameScreen.TEXTURE_PIECES_HEIGHT,
+                ((transparent ? 0x7F : 0xFF) << 24) | 0xFFFFFF // if I wanted, it could be (-1 >>> (transparent ? 1 : 0))
```

Comment:
```suggestion
                transparent ? 0x7f_ffffff : 0xff_ffffff
```
Also no need to point out the more esoteric way of doing it if you're not actually using it.